sklearn/ensemble/iforest.py

```diff
@@ -206,7 +219,9 @@ def predict(self, X):
 
         depths += _average_path_length(n_samples_leaf)
 
-        if not isinstance(self.max_samples, (numbers.Integral, np.integer)):
```

Comment:
And in that case you'll need to duplicate code as you've done before to set `max_samples_` in fit() I'm affraid.